java/vortex-jni/src/test/java/dev/vortex/api/expressions/proto/TestOpenErrors.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.api.expressions.proto;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import dev.vortex.jni.NativeFileMethods;
+import org.junit.jupiter.api.Test;
+
+public final class TestOpenErrors {
+    @Test
+    public void testOpenThrows() {
+        RuntimeException runtimeException = assertThrows(RuntimeException.class, () -> {
+            NativeFileMethods.open("bad_scheme:///fake-location", null);
+        });
+        assertTrue(runtimeException.getMessage().contains("Invalid URL"));
```

Comment:
object_store Error return turns into a Java RuntimeException via the machinery in `try_or_throw` (which is heavily inspired by the [Comet error handling](https://github.com/apache/datafusion-comet/blob/main/native/core/src/errors.rs)